code/__DEFINES/spatial_gridmap.dm

```diff
@@ -1,12 +1,13 @@
-///each cell in a spatial_grid is this many turfs in length and width
+/// each cell in a spatial_grid is this many turfs in length and width (with world.max(x or y) being 255, 15 of these fit on each side of a z level)
 #define SPATIAL_GRID_CELLSIZE 17
-///Takes a coordinate, and spits out the spatial grid index (x or y) it's inside
+/// Takes a coordinate, and spits out the spatial grid index (x or y) it's inside
 #define GET_SPATIAL_INDEX(coord) ROUND_UP((coord) / SPATIAL_GRID_CELLSIZE)
-#define GRID_INDEX_TO_COORDS(index) (index * SPATIAL_GRID_CELLSIZE)
+/// changes the cell_(x or y) vars on /datum/spatial_grid_cell to the x or y coordinate on the map for the LOWER LEFT CORNER of the grid cell.
+/// index is from 1 to SPATIAL_GRID_CELLS_PER_SIDE
+#define GRID_INDEX_TO_COORDS(index) ((((index) - 1) * SPATIAL_GRID_CELLSIZE) + 1)
```

Comment:
the math was wrong and had an off-by-one error on both x and y axes so it gave the coordinates of the upper right corner of the grid cell, now its lower left

Reply:
off by one moment